### ISSUE TITLE
Sandbox is recreated each time the runIdea task is run

### ISSUE THREAD

**Issue by pavel-v-chernykh** [alexeykudinkin/intellij-rust#106]
As discussed at project gitter chat make something with the rebuilding of the plugin sandbox.

> Hi, folks. I have an issue with ```./gradlew runIdea``` command. It is constantly refreshing plugin sandbox folder. It causes me to install PsiViewer plugin all the time I do some changes and run ```./gradlew runIdea``` command. Does somebody know how to automate plugin installation when I run ```./gradlew runIdea```?

May be through additional task which will download plugins, apply IDE settings and so on.

**Comment by Turbo87** [comment]
as mentioned in the chat: it works fine over here unless I specifically run the `clean` task. I'm not quite sure what I'm doing differently...

**Comment by pavel-v-chernykh** [comment]
You need to make changes in sources before second run. runIdea will recreate sandbox if only it detects changes.

**Comment by Turbo87** [comment]
you mean changes in the grammar file?

**Comment by Turbo87** [comment]
hmm, still no problem even after changing the grammar... what platform are you on?

**Comment by Turbo87** [comment]
@pavel-v-chernykh could you attach the output of running `./gradlew runIdea` which then recreates the sandbox? I'd like to see what tasks it actually runs

**Comment by pavel-v-chernykh** [comment]
@Turbo87 yeah, there is the log (run without changes)
```
:generateRustLexer UP-TO-DATE
:generateTomlLexer UP-TO-DATE
:generateLexers UP-TO-DATE
:generateRustPsiAndParser UP-TO-DATE
:generateTomlPsiAndParser UP-TO-DATE
:generateParsers UP-TO-DATE
:generate UP-TO-DATE
:compileKotlin UP-TO-DATE
:compileJava UP-TO-DATE
:processResources
:classes
:patchPluginXml
:prepareSandbox UP-TO-DATE
:runIdea
```

and another one when I introduce some changes in any of *.kt

```
:generateRustLexer UP-TO-DATE
:generateTomlLexer UP-TO-DATE
:generateLexers UP-TO-DATE
:generateRustPsiAndParser UP-TO-DATE
:generateTomlPsiAndParser UP-TO-DATE
:generateParsers UP-TO-DATE
:generate UP-TO-DATE
:compileKotlin
w: /Users/pavel/Projects/intellij-rust/src/org/rust/lang/core/parser/RustParserUtil.kt: (38, 17): Variable 'r' initializer is redundant
w: /Users/pavel/Projects/intellij-rust/src/org/rust/lang/core/parser/RustParserUtil.kt: (65, 50): Parameter 'level' is never used
w: /Users/pavel/Projects/intellij-rust/src/org/rust/lang/core/parser/RustParserUtil.kt: (68, 49): Parameter 'level' is never used
w: /Users/pavel/Projects/intellij-rust/src/org/rust/lang/core/parser/RustParserUtil.kt: (87, 44): Parameter 'level' is never used
w: /Users/pavel/Projects/intellij-rust/src/org/rust/lang/core/parser/RustParserUtil.kt: (99, 45): Parameter 'level' is never used
:compileJava
[ant:instrumentIdeaExtensions] : warning: 'includeantruntime' was not set, defaulting to build.sysclasspath=last; set to false for repeatable builds
:processResources
:classes
:patchPluginXml
:prepareSandbox
:runIdea
```

looks like ```:prepareSandbox UP-TO-DATE``` vs ```:prepareSandbox``` matters

**Comment by Turbo87** [comment]
have a look at https://github.com/JetBrains/gradle-intellij-plugin#configuration. maybe setting `sandboxDirectory` to some place outside of the `build` folder will help?

**Comment by Turbo87** [comment]
after modifying a `.kt` file here:

````
:generateRustLexer UP-TO-DATE
:generateTomlLexer UP-TO-DATE
:generateLexers UP-TO-DATE
:generateRustPsiAndParser UP-TO-DATE
:generateTomlPsiAndParser UP-TO-DATE
:generateParsers UP-TO-DATE
:generate UP-TO-DATE
:compileKotlin
w: /Users/tbn/Code/intellij-rust/src/org/rust/lang/core/parser/RustParserUtil.kt: (38, 17): Variable 'r' initializer is redundant
w: /Users/tbn/Code/intellij-rust/src/org/rust/lang/core/parser/RustParserUtil.kt: (65, 50): Parameter 'level' is never used
w: /Users/tbn/Code/intellij-rust/src/org/rust/lang/core/parser/RustParserUtil.kt: (68, 49): Parameter 'level' is never used
w: /Users/tbn/Code/intellij-rust/src/org/rust/lang/core/parser/RustParserUtil.kt: (87, 44): Parameter 'level' is never used
w: /Users/tbn/Code/intellij-rust/src/org/rust/lang/core/parser/RustParserUtil.kt: (99, 45): Parameter 'level' is never used
:compileJava UP-TO-DATE
:processResources
:classes
:patchPluginXml
:prepareSandbox UP-TO-DATE
:runIdea
````

note that my `prepareSandbox` task is still `UP-TO-DATE`

**Comment by Turbo87** [comment]
@zolotov do you have any idea what's going on here?

**Comment by Turbo87** [comment]
`./gradlew runIdea --debug` with modified `.kt` file:

```
09:44:53.350 [INFO] [org.gradle.execution.taskgraph.AbstractTaskPlanExecutor] :prepareSandbox (Thread[Daemon worker Thread 2,5,main]) started.
09:44:53.350 [LIFECYCLE] [class org.gradle.TaskExecutionLogger] :prepareSandbox
09:44:53.350 [DEBUG] [org.gradle.api.internal.tasks.execution.ExecuteAtMostOnceTaskExecuter] Starting to execute task ':prepareSandbox'
09:44:53.351 [DEBUG] [org.gradle.api.internal.artifacts.ivyservice.resolveengine.DefaultArtifactDependencyResolver] Resolving configuration ':runtime'
09:44:53.353 [DEBUG] [org.gradle.api.internal.artifacts.ivyservice.ivyresolve.memcache.InMemoryCachedRepositoryFactory] Reusing in-memory cache for repo 'MavenRepo' [4471c3b2f5ea2d40ffaa8b3948cdbacd].
09:44:53.354 [DEBUG] [org.gradle.api.internal.artifacts.ivyservice.ivyresolve.memcache.InMemoryCachedRepositoryFactory] Reusing in-memory cache for repo 'maven' [7e15349dd3cb7cecebd619836ced62a4].
09:44:53.355 [DEBUG] [org.gradle.api.internal.artifacts.ivyservice.ivyresolve.memcache.InMemoryCachedRepositoryFactory] Reusing in-memory cache for repo 'ivy' [54ff0ef4b43acbbea32d9d404ff114d8].
09:44:53.356 [DEBUG] [org.gradle.api.internal.artifacts.ivyservice.resolveengine.graph.DependencyGraphBuilder] Visiting configuration :intellij-rust:0.0.1.SNAPSHOT(runtime).
09:44:53.356 [DEBUG] [org.gradle.api.internal.artifacts.ivyservice.resolveengine.graph.DependencyGraphBuilder] Visiting dependency :intellij-rust:0.0.1.SNAPSHOT(runtime) -> org.jetbrains.kotlin:kotlin-stdlib:1.0.0-beta-1103(compile)
09:44:53.356 [DEBUG] [org.gradle.api.internal.artifacts.ivyservice.resolveengine.graph.DependencyGraphBuilder] Selecting new module version org.jetbrains.kotlin:kotlin-stdlib:1.0.0-beta-1103
09:44:53.356 [DEBUG] [org.gradle.api.internal.artifacts.ivyservice.ivyresolve.RepositoryChainComponentMetaDataResolver] Attempting to resolve component for org.jetbrains.kotlin:kotlin-stdlib:1.0.0-beta-1103 using repositories [MavenRepo, maven, ivy]
09:44:53.357 [DEBUG] [org.gradle.api.internal.artifacts.ivyservice.ivyresolve.RepositoryChainComponentMetaDataResolver] Using org.jetbrains.kotlin:kotlin-stdlib:1.0.0-beta-1103 from Maven repository 'MavenRepo'
09:44:53.357 [DEBUG] [org.gradle.api.internal.artifacts.ivyservice.resolveengine.graph.DependencyGraphBuilder] Visiting dependency :intellij-rust:0.0.1.SNAPSHOT(runtime) -> com.jetbrains:ideaIC:15.0(compile)
09:44:53.357 [DEBUG] [org.gradle.api.internal.artifacts.ivyservice.resolveengine.graph.DependencyGraphBuilder] Selecting new module version com.jetbrains:ideaIC:15.0
09:44:53.358 [DEBUG] [org.gradle.api.internal.artifacts.ivyservice.ivyresolve.RepositoryChainComponentMetaDataResolver] Attempting to resolve component for com.jetbrains:ideaIC:15.0 using repositories [MavenRepo, maven, ivy]
09:44:53.358 [DEBUG] [org.gradle.api.internal.artifacts.ivyservice.ivyresolve.RepositoryChainComponentMetaDataResolver] Using com.jetbrains:ideaIC:15.0 from Ivy repository 'ivy'
09:44:53.358 [DEBUG] [org.gradle.api.internal.artifacts.ivyservice.resolveengine.graph.DependencyGraphBuilder] Visiting dependency :intellij-rust:0.0.1.SNAPSHOT(runtime) -> com.jetbrains:ideaIC:15.0(runtime)
09:44:53.358 [DEBUG] [org.gradle.api.internal.artifacts.ivyservice.resolveengine.graph.DependencyGraphBuilder] Visiting configuration org.jetbrains.kotlin:kotlin-stdlib:1.0.0-beta-1103(default).
09:44:53.359 [DEBUG] [org.gradle.api.internal.artifacts.ivyservice.resolveengine.graph.DependencyGraphBuilder] Visiting dependency org.jetbrains.kotlin:kotlin-stdlib:1.0.0-beta-1103(default) -> org.jetbrains.kotlin:kotlin-runtime:1.0.0-beta-1103(compile,runtime)
09:44:53.359 [DEBUG] [org.gradle.api.internal.artifacts.ivyservice.resolveengine.graph.DependencyGraphBuilder] Selecting new module version org.jetbrains.kotlin:kotlin-runtime:1.0.0-beta-1103
09:44:53.359 [DEBUG] [org.gradle.api.internal.artifacts.ivyservice.ivyresolve.RepositoryChainComponentMetaDataResolver] Attempting to resolve component for org.jetbrains.kotlin:kotlin-runtime:1.0.0-beta-1103 using repositories [MavenRepo, maven, ivy]
09:44:53.359 [DEBUG] [org.gradle.api.internal.artifacts.ivyservice.ivyresolve.RepositoryChainComponentMetaDataResolver] Using org.jetbrains.kotlin:kotlin-runtime:1.0.0-beta-1103 from Maven repository 'MavenRepo'
09:44:53.360 [DEBUG] [org.gradle.api.internal.artifacts.ivyservice.resolveengine.graph.DependencyGraphBuilder] Visiting configuration com.jetbrains:ideaIC:15.0(compile).
09:44:53.360 [DEBUG] [org.gradle.api.internal.artifacts.ivyservice.resolveengine.graph.DependencyGraphBuilder] Visiting configuration com.jetbrains:ideaIC:15.0(runtime).
09:44:53.360 [DEBUG] [org.gradle.api.internal.artifacts.ivyservice.resolveengine.graph.DependencyGraphBuilder] Visiting configuration org.jetbrains.kotlin:kotlin-runtime:1.0.0-beta-1103(compile).
09:44:53.360 [DEBUG] [org.gradle.api.internal.artifacts.ivyservice.resolveengine.graph.DependencyGraphBuilder] Visiting configuration org.jetbrains.kotlin:kotlin-runtime:1.0.0-beta-1103(master).
09:44:53.361 [DEBUG] [org.gradle.api.internal.artifacts.ivyservice.resolveengine.graph.DependencyGraphBuilder] Visiting configuration org.jetbrains.kotlin:kotlin-runtime:1.0.0-beta-1103(runtime).
09:44:53.363 [DEBUG] [org.gradle.api.internal.artifacts.ivyservice.resolveengine.oldresult.TransientConfigurationResultsBuilder] Flushing resolved configuration data in Binary store in /private/var/folders/94/9qmjxm8x1332918s1nczny8w0000gn/T/gradle7012879648662622741.bin. Wrote root :intellij-rust:0.0.1.SNAPSHOT:runtime.
09:44:53.369 [DEBUG] [org.gradle.api.internal.artifacts.repositories.resolver.DefaultExternalResourceArtifactResolver] Loading file:/Users/tbn/.gradle/caches/modules-2/files-2.1/com.jetbrains.intellij.idea/ideaIC/15.0/749217a0fb59c90a54e0a1adb4d9ab0ea8871c4/ideaIC-15.0/com.jetbrains/ideaIC/15.0/tools-15.0.jar
09:44:53.370 [DEBUG] [org.gradle.api.internal.artifacts.repositories.resolver.DefaultExternalResourceArtifactResolver] Loading file:/Users/tbn/.gradle/caches/modules-2/files-2.1/com.jetbrains.intellij.idea/ideaIC/15.0/749217a0fb59c90a54e0a1adb4d9ab0ea8871c4/ideaIC-15.0/com.jetbrains/ideaIC/15.0/tools-intellij-rust.jar
09:44:53.370 [DEBUG] [org.gradle.api.internal.artifacts.repositories.resolver.DefaultExternalResourceArtifactResolver] Loading file:/Users/tbn/.gradle/caches/modules-2/files-2.1/com.jetbrains.intellij.idea/ideaIC/15.0/749217a0fb59c90a54e0a1adb4d9ab0ea8871c4/ideaIC-15.0/tools.jar
09:44:53.370 [DEBUG] [org.gradle.api.internal.artifacts.repositories.resolver.DefaultExternalResourceArtifactResolver] Loading file:/Library/Java/JavaVirtualMachines/jdk1.8.0_66.jdk/Contents/Home/lib/tools.jar
09:44:53.373 [DEBUG] [org.gradle.api.internal.tasks.execution.SkipUpToDateTaskExecuter] Determining if task ':prepareSandbox' is up-to-date
09:44:53.474 [INFO] [org.gradle.api.internal.tasks.execution.SkipUpToDateTaskExecuter] Skipping task ':prepareSandbox' as it is up-to-date (took 0.101 secs).
09:44:53.474 [DEBUG] [org.gradle.api.internal.tasks.execution.ExecuteAtMostOnceTaskExecuter] Finished executing task ':prepareSandbox'
09:44:53.474 [LIFECYCLE] [class org.gradle.TaskExecutionLogger] :prepareSandbox UP-TO-DATE
09:44:53.475 [INFO] [org.gradle.execution.taskgraph.AbstractTaskPlanExecutor] :prepareSandbox (Thread[Daemon worker Thread 2,5,main]) completed. Took 0.125 secs.
```

**Comment by zolotov** [comment]
@Turbo87 thanks for mention me. Yes I know what's going on, filed an issue in gradle-intellij-plugin

**Comment by Turbo87** [comment]
@zolotov awesome, thanks

**Comment by alexeykudinkin** [comment]
Fixed by extraction of the IDEA's sandbox out of `build` directory.

**Comment by zolotov** [comment]
Fixed in gradle-plugin 0.0.29
